### ISSUE TITLE
question about rate loss

### ISSUE THREAD

**Issue by jinhaohust** [Nikolai10/MRIC#1]
In the paper, it is mentioned that the rate parameter \( r(\hat{y}) \) is used to constrain the bitrate, similar to the method used in HiFiC, when designing the rate hyperparameter in the loss function:

"We note that in contrast to Eq. 1, we have the rate parameter \(\lambda'\) on \( r(\hat{y}) \) instead of MSE. This formulation allows us to target different bitrates without changing the relative weight of the distortion compared to other terms."

However, while reading the code, I found that the \(\lambda\) used is a fixed value and does not employ the trick mentioned in the paper:

```python
weighted_rate_loss = self.lmbda * cfg_mric.lmbda_weight * total_bpp
```

I would like to ask: Does the trick of constraining the bitrate using \( r(\hat{y}) \) have no effect on the Multi-Realism model?
thanks for your reply！

**Comment by Nikolai10** [comment]
Dear @jinhaohust, 

I am not sure, if I understand your question correctly.

From https://arxiv.org/pdf/2212.13824, Section 3.2, we see that λ'=100λ to achieve different bitrates.

This is exactly what we implemented: in our code, λ' is represented as `self.lmbda * cfg_mric.lmbda_weight`, with [cfg_mric.lmbda_weight = 100](https://github.com/Nikolai10/MRIC/blob/master/src/config.py#L20C5-L20C17). Note that only `cfg_mric.lmbda_weight` is a fixed value, `self.lmbda` is chosen from {0.001, 0.002, 0.008, 0.032, 0.128, 0.512} during training and set as [command line argument](https://github.com/Nikolai10/MRIC/blob/master/src/amtm2023.py#L832).

total_bpp corresponds to the rate estimate r(y_hat).

Does this make sense?

Kind regards,
Nikolai

**Comment by jinhaohust** [comment]
Thank you very much for your response. I realized I made a mistake. I initially thought that the MR used the method from HiFiC to constrain the bit rate, specifically with r_t using two hyperparameters, lambda_a and lambda_b. After reading your reply, I understand that I was mistaken and that you are correct. I greatly appreciate your prompt response！！
